python/knowhere/knowhere.i

```diff
@@ -354,4 +356,51 @@ DumpRangeResultDis(knowhere::DataSetPtr result, float* dis, int len) {
     }
 }
 
+void
+Dump(knowhere::BinarySetPtr binset, const std::string& file_name) {
+    auto binary_set = *binset;
+    auto binary_map = binset -> binary_map_;
+    std::ofstream outfile;
+    outfile.open(file_name, std::ios::out | std::ios::trunc);
+    for (auto it = binary_map.begin(); it != binary_map.end(); ++it) {
+        // serialization: name_length(size_t); name(char[]); binset_size(size_t); binset(uint8[]); 
+        auto name = it->first;
+        outfile << name.size();
+        outfile << name;
+        auto value = it->second;
+        outfile << value->size;
+        outfile.write(reinterpret_cast<char*>(value->data.get()), value->size);
+    }
+    // end with 0
+    outfile << 0;
+    outfile.close();
+}
+
+void
+Load(knowhere::BinarySetPtr binset, const std::string& file_name) {
+    std::ifstream infile;
+    infile.open(file_name, std::ios::in);
+    size_t name_len;
+    char flag;
+    while (true) {
+        // serialization: name_length(size_t); name(char[]); binset_size(size_t); binset(uint8[]); 
+        infile >> name_len;
+        if (name_len == 0) break;
+
+        auto _name = new char[name_len];
+        infile.read(_name, name_len);
+        std::string name(_name, name_len);
+
+        int64_t size;
+        infile >> size;
+
+        auto data = new uint8_t[size];
```

Comment:
any checks if the size is too large, negative or making no sense? :)

Reply:
negative checked.
how to check for too large size? or we can just wait for the program to report an exception?